src/LSST/control/PowerSubsystem.cpp

```diff
@@ -404,27 +404,27 @@ PowerSubsystem::PowerState PowerSubsystem::getTargPowerState() const {
 }
 
 
-void PowerSubsystem::setPowerOn() {
+bool PowerSubsystem::setPowerOn() {
     VMUTEX_NOT_HELD(_powerStateMtx);
     auto outputPort = _fpgaIo->getOutputPort();
     lock_guard<util::VMutex> lg(_powerStateMtx);
-    _setPowerOn();
+    return _setPowerOn();
 }
 
-void PowerSubsystem::_setPowerOn() {
+bool PowerSubsystem::_setPowerOn() {
     VMUTEX_HELD(_powerStateMtx);
 
     if (_checkForFaults()) {
         LERROR(getClassName(), " _setPowerOn cannot turn on due to faults");
-        _sendFaultMgrError(500003, "Internal ERROR: Faults preventing operation to proceed");
+        faultmgr::FaultMgr::get().faultMsg(500003, "Internal ERROR: Faults preventing operation to proceed");
```

Comment:
For the error cord such as `500003` or others, it would be better to read/define from a configuration file. For example:
https://github.com/lsst-ts/ts_mtm2_cell/blob/develop/config/User-errors.txt

Reply:
I made jira ticket DM-41195 use that configuration file for error numbers and messages.